tests/test_experiment.py

```diff
@@ -194,3 +194,22 @@ def test_participant_task_completed_skips_assigning_qualification_if_overrecruit
         exp.participant_task_completed(participant)
 
         participant.recruiter.assign_experiment_qualifications.assert_not_called()
+
+
+class TestTaskRegistration(object):
+    def test_deferred_task_decorator(self, cleared_tasks):
+        from dallinger.experiment import scheduled_task
+
+        decorator = scheduled_task("interval", minutes=15)
+        assert len(cleared_tasks) == 0
+
+        def fake_task():
+            pass
+
+        # Decorator does not modify or wrap the function
+        assert decorator(fake_task) is fake_task
+        assert len(cleared_tasks) == 1
```

Comment:
I was super confused here, and I realized it's just the fixture name. How about `tasks_with_cleanup` or something to suggest that the "clearing" as an aspect of fixture teardown?

Reply:
Sure